### PR TITLE
[VTA][Refactor] Introducing VTA_HW_PATH for easier migration

### PR DESCRIPTION
**Pull request by tmoreau89**
TVM community has voted for the `vta/vta-hw` directory migration (see https://github.com/apache/incubator-tvm/issues/5102 for context). To make the switch easier:

1.  We're making it easier to migrate that sub-directory with a `VTA_HW_PATH` environment variable
2. To run unit testing in the new `vta-hw` repo, we're introducing the `TVM_PATH` environment variable to point to TVM repo
3. Updated the instructions to reflect that change
4. Turned off building tsim/fsim libraries for VTA by default so average TVM users aren't asked to set these paths
5. Remove VTA sources in `vta-hw` from doxygen -- `vta-hw` will have it's own doxygen generated docs
6. Migrated pkg_config.py to `vta-hw` because it makes more sense to couple that with the config files.

### CODE REVIEW

**Comment by tmoreau89** [comment]
@tqchen @liangfu

File: vta/vta-hw/config/pkg_config.py [hunk 1]
```diff
@@ -22,6 +22,20 @@
 
 import json
 import glob
+import os
+
+def get_vta_hw_path():
+    """Get the VTA HW path."""
```
Comment (tqchen) [inline vta/vta-hw/config/pkg_config.py]: It is still better to have a default

File: vta/vta-hw/config/pkg_config.py [hunk 2]
```diff
@@ -22,6 +22,20 @@
...
+    VTA_HW_PATH = os.getenv('VTA_HW_PATH', None)
+    assert VTA_HW_PATH, "Please set VTA_HW_PATH to point to TVM dir \
+                         VTA subdirectory (<tvm-root>/vta/vta-hw)."
+    return VTA_HW_PATH
+
+def get_tvm_path():
+    """Get the TVM path."""
+    TVM_PATH = os.getenv('TVM_PATH', None)
```
Comment (tqchen) [inline vta/vta-hw/config/pkg_config.py]: it is better to have a default

File: vta/vta-hw/config/pkg_config.py (hunk 1 above)
Comment (tmoreau89) [reply vta/vta-hw/config/pkg_config.py]: thanks I've addressed them

File: vta/vta-hw/config/pkg_config.py (hunk 2 above)
Comment (tmoreau89) [reply vta/vta-hw/config/pkg_config.py]: yes, I'll set one

**Comment by zhiics** [comment]
Seems master CI is broken now.

**Comment by tqchen** [comment]
@zhiics https://github.com/apache/incubator-tvm/pull/5165 should fix the issue

**Pull request merged**